packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.test.tsx

```diff
@@ -17,21 +17,13 @@ describe('ProductEditForm', () => {
         />
       );
 
-      await waitUntilTableIsLoaded();
-
-      const saveButton = screen.getByRole('button', {
+      const saveButton = screen.findByRole('button', {
```

Comment:
I think you could make it in a more common way to change it into:

```ts
const saveButton = await screen.findByRole(...);

await waitFor(() => expect(saveButton).not.toBeDisabled());
```

Reply:
Example: https://www.notion.so/monite/How-to-write-Unit-Tests-for-Frontend-ea3172219c234127bf15701a382bc0b8?pvs=4#65b7ade1feaf4e0785db78c361120217

Reply:
I have an alternative view of this flow:

```ts
// Start by finding the button
const saveButton = screen.findByRole('button', { name: 'Update' });

// If the button is not found, throw a conventional Jest error message with IDE in-place highlighting
await expect(saveButton).resolves.toBeInTheDocument();

// Using one more `.resolves.` here is not a big deal in my opinion
await waitFor(() => expect(saveButton).resolves.not.toBeDisabled());

// Finally, click the already resolved `saveButton`
fireEvent.click(await saveButton);
```